mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -236,6 +236,18 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
   private def getDistinctSplits(splits: Array[Double]): Array[Double] = {
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
+
+    // If array contains both -0.0 and 0.0, current array.distinct is buggy, the distinct result
+    // may contain both -0.0 and 0.0 or only contain -0.0 or 0.0, this is because:
+    //  -0.0 and 0.0 has different hashCode but they are equal, this break the contract between
+    // equals() and hashCode(): If two objects are equal, then they must have the same hash code.
```

Comment:
Could you update the comment per https://github.com/scala/bug/issues/11994#issuecomment-627070712?